modules/ruxrand/src/spin_rand.rs

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) [2023] [Syswonder Community]
+ *   [Ruxos] is licensed under Mulan PSL v2.
+ *   You can use this software according to the terms and conditions of the Mulan PSL v2.
+ *   You may obtain a copy of Mulan PSL v2 at:
+ *               http://license.coscl.org.cn/MulanPSL2
+ *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ *   See the Mulan PSL v2 for more details.
+ */
+
+use core::{
+    fmt::Debug,
+    ops::RangeInclusive,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use rand::{Rng, RngCore};
+use spinlock::SpinRandIf;
```

Comment:
`SpinRandIf` should be move into crate `ruxrand` and try to aviod use `crate_interface::impl_interface`. if ruxrand and spinlock must depend on each other, it seems to be a little meaningless to divide them as two crates.

Reply:
it looks like spinlock has dependency on ruxrand, but ruxrand no necassarily dependency on spinlock. You can try to define SpinRandIf in ruxrand directly.

Reply:
You're right. I moved `ExpRand` into `ruxrand` and removed `SpinRandIf`.